### PR TITLE
Add colors and formatting to dispatch-cli

### PR DESCRIPTION
**Pull request by Marenz**
color-dispatch

### CODE REVIEW

File: src/frequenz/client/dispatch/__main__.py [hunk 1]
```diff
@@ -31,10 +25,140 @@
...
+    """Return the name of the month."""
+    return datetime(2000, month, 1).strftime("%B")
+
+
+# pylint: disable=too-many-statements, too-many-locals
+def print_dispatch(dispatch: Dispatch) -> None:
+    """Print the dispatch details in a nicely formatted way with colors."""
+    # Determine the status and color
+    status: str = "running" if dispatch.started else "not running"
+    status_color: str = "green" if dispatch.started else "red"
```
Comment (llucax) [inline src/frequenz/client/dispatch/__main__.py:74]: Nitpick: What about gray for not started? Red looks like an error to me. Maybe we can sort of mimic `systemctl status` as many users may be familiar with them?

File: src/frequenz/client/dispatch/__main__.py [hunk 2]
```diff
@@ -31,10 +25,140 @@
...
+    ]
+    max_key_length: int = max(len(k) for k in keys)
+
+    # Helper function to format each line
+    def format_line(key: str, value: str, color: str = "cyan") -> str:
+        key_str = click.style(f"{key}:", fg=color)
+        val_color = "white"
+
+        if value in ("None", "False"):
+            val_color = "red"
```
Comment (llucax) [inline src/frequenz/client/dispatch/__main__.py:120]: Also here maybe gray for `None`?

File: src/frequenz/client/dispatch/__main__.py [hunk 3]
```diff
@@ -52,8 +176,15 @@
     show_envvar=True,
     required=True,
 )
+@click.option(
+    "--raw",
```
Comment (llucax) [inline src/frequenz/client/dispatch/__main__.py:180]: Possible future improvements:
* Split `--raw` from `--no-color`
* Set color on/off automatically based on `sys.stdin.isatty()`

**Review by llucax — APPROVED** [review]
Awesome! It might also be a good idea to maybe set constants with colors, so they are easier to tune if needed, like `COLOR_DISPATCH_STARTED`, `VALUE_NONE_COLOR`, `COLOR_VALUE_TRUE`, etc.

But all comments optional.

File: src/frequenz/client/dispatch/__main__.py (hunk 3 above)
Comment (Marenz) [reply src/frequenz/client/dispatch/__main__.py:180]: > Set color on/off automatically based on sys.stdin.isatty()

that actually seems to happen automatically, at least there is no color if I pipe it through cat or forward it into a file

File: src/frequenz/client/dispatch/__main__.py (hunk 1 above)
Comment (Marenz) [reply src/frequenz/client/dispatch/__main__.py:74]: My desire was that it stands out immediately what the status is :)

File: src/frequenz/client/dispatch/__main__.py (hunk 3 above)
Comment (llucax) [reply src/frequenz/client/dispatch/__main__.py:180]: Ah cool. Extra point for `click` 💯

**Pull request merged**